### PR TITLE
List Windows XP as Tier 2

### PR DESCRIPTION
**Pull request by lambda**
Based on the definitions of the three tiers of support, it appears that
Windows XP is actually Tier 2.  Compatibility with Windows XP (and
Vista) is done by dispatching a few Windows API wrappers at runtime to
either the real calls if available, or a fallback stub if not, that
either panics or returns an "unimplemented" status code.  According to
the definitions of the tiers, Tier 2 means that libstd builds but is not
tested, and release artifacts are provided.  Since the existing release
artifacts for Windows 7+ should work on Windows XP+ due to the
compatibility fallbacks, Windows XP support should be listed in Tier 2.

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks for the PR! I'm a little wary to promote this to tier 2 just yet, though. To me at least "compiles" includes at least "hello world links successfully", but unfortunately we're not providing that guarantee just yet. We have in the past updated the standard library to not literally link to Vista+ APIs and instead only attempt to probe them at runtime, but we don't have a guarantee that no others have leaked in. We do currently strive for our release artifacts to be compatible with Windows XP, so we certainly have that half, of course!

Basically, my opinion here is that XP will become a tier 2 platform once we've got automation that a hello world binary targeting Windows XP can be created, but before that I'd be hesitant to make such a guarantee.

**Comment by alexcrichton** [comment]
Closing due to inactivity, but we can be sure to list XP under tier 2 once we start regularly testing that it at least compiles!
